### ISSUE TITLE
Support for running multiple Intern instances in a single Grunt run

### ISSUE THREAD

**Issue by rhpijnacker** [theintern/intern#894]
In my grunt configuration, I have split the tasks so I can run part of my tests with chrome, safari and ie11 and part with only chrome and safari.
When I run both tasks in one grunt run, the first completes succesfully. The second one, gives an error at the start:
```
Running "intern:batch2" (intern) task
Error: Loader script node_modules/intern/loaders/dojo.js did not register a loader callback
  at <src\lib\executors\Executor.ts:915:12>
  at <node_modules\@dojo\core\async\Task.ts:351:12>
  at handler  <node_modules\@dojo\core\async\ExtensiblePromise.ts:229:14>
Warning: Loader script Platform/WebApps/ISCV/interntests/dojoloader.js did not register a loader callback Use --force to continue.
```
The reason seems to be that (since the loader is already required once) the Executor does not get the right trigger that the loader finished loading.

Intern version: 4.1.5
Node version: 6.11
NPM version: 5.2
Grunt version: 1.0.1

**Comment by jason0x43** [comment]
Hmmm... That may require an update to the loader script API so that Intern can teardown the loaders when it finishes a run. That, or loader scripts can check for the existence of their loader during initialization and reuse or replace it (probably replace).

**Comment by rhpijnacker** [comment]
At first I thought the latter would be trivial to implement, until I realized that the loader script is actually just called only once. So "loader scripts can check for the existence of their loader during initialization" would not actually work. This should be handled in `Executor`, since it is the place that loads the scripts so can administrate whether a loader was already loaded.

**Comment by jason0x43** [comment]
> ...the loader script is actually just loaded and executed only once.

That, I think, is the real issue. The first time Intern is run, the loader script is loaded and it registers a loader callback. The second time Intern is run _in the same Node session_, the loader script is required but it isn’t actually re-loaded or evaluated since it’s already cached by the Node loader, so its registration function isn’t called the second time.

There are a couple ways to try to deal with this. One is for Intern to keep track of individual modules it loaded and then de-register them at the end of the session. This process could potentially be simplified a bit by adding a ‘deregistration’ function to the loader and plugin APIs that Intern could use to unload them after a run.

Another (and probably much easier in the long run) would be for Intern to load everything it needs to in a sandbox that could simply be discarded after a test run.

**Comment by rhpijnacker** [comment]
Do you have a suggestion for what that deregistration function should look like?

**Comment by jason0x43** [comment]
Actually, the simplest immediate solution would probably be to run intern in a subprocess rather than loading it into Grunt's context (which is what the Intern 3 grunt task did).

**Comment by jason0x43** [comment]
This is also related to #919, which is a more general request to be able to reuse an executor.

**Comment by jason0x43** [comment]
Closing this in favor of #919. Resolving that one should automatically take care of this.
